Document date filter relocation in Reports (#753)
mautic/user-documentation: docs/reports/reports.rst

Reviewer: ```suggestion
|

* **Columns** - Select the columns of data that you want to appear in the table of data in the Report. Click the column name in the left column to have it show in the Report. You can remove a column from the Report by clicking on its right column. The column returns to its original position on the left side. For example, if you select ID, it refers to the **ID** for the parent data source Category that you selected earlier Contact ID, Email ID, Asset ID, etc. It's recommended that you include **ID** in all Reports.
* **Order** - For sorting the data in the Report, select data Points. The available sort options are **Ascending** and **Descending**. To add multiple columns, click **Add Order**. For fields that use text, an **Ascending** order lists values starting with B after values starting with A and so on. For number or date fields, the higher the number or later the date, the lower on the list the row is. Descending order is the opposite.

  .. note:: 

     Adding multiple fields to order by uses the last one in the Order list first. Ordering by **First Name Ascending** and adding **Email Ascending**, for instance, sorts the Email column first and duplicate rows are then sorted by first name ascending.

* **Filters** -  Filter the data using conditions and values. This allows the generation of very granular Reports. This option helps you to narrow down the data included in the Report. The data Points used for filters don’t have to be columns that appear in the Report table. A commonly used filter for any Reports that include Contact record data is Email Not Empty, which displays only identified Contacts in the Report. Additional use cases can include Contacts or items that match a certain value, events happening within a certain date range, etc.

  .. note::

     .. vale off

     Setting **Dynamic** to **Yes** allows Users to change the data they see in a Report without editing it. They can access the filter from the **Filters** panel at the top of the Report page. The Filters panel only appears when a Report has at least one dynamic filter configured.

     .. vale on

* **Group by** - Select the columns for which you want to group data. By default, Reports show all items individually. In many Reports, you may see the same Contact, Company, or item appear multiple times. To only see each record listed once, you can add a grouping based on some attribute for the record. 

  You can use Email or Contact ID to display a single row per Contact record. For example, you can group by **Contact ID** to view the unique number of Asset Downloads or Form Submissions for a single Form, instead of total Asset Downloads or Form Submissions, which could include duplicates.

* **Calculated columns** - Select the function that you want to apply to individual columns. Calculated columns display count, average, sum, or the minimum or maximum values from a selected field. They're only available when using a grouping to show a calculation for that grouping. Continuing with the previous example of grouping by a Contact ID number or Email address, a ``COUNT`` calculation displays how many times that Contact record appears on the Report if not for the grouping.
```

     * Multi Touch Attributions
     * First Touch Attributions
     * Last Touch Attributions
     * Contact Point Log
     * Frequency Rules
     * Segment Membership
     * Do Not Contact
     * UTM Codes
     * Group score

  #. :doc:`Companies</companies/companies_overview>`

     * Companies

  #. :doc:`Mobile Notifications</channels/marketing_messages>`

     * Mobile Notifications
     * Mobile Notifications Sent

  #. :doc:`Pages</components/landing_pages>`

     * Landing Pages
     * Page hits

  #. Videos

     * Video hits

  As demonstrated, Mautic displays the data sources in the format: ``<Parent data source> <Child data source>``

  The parent data source provides a high-level summary of the data while the child data source provides a more granular view of Contact behavior that includes your Custom Fields and values.

* **Description - optional** - Add detailed information about the Report, helping you and other Users better understand what the Report is for. You may want to include more information about filters, people, or departments.
* **Published** - Select **Yes** to ensure that the Report is active, ready to send when scheduled in the Report options.

> * **Visible for all logged-in Users** - Select **Yes** so that all Users can access the Report. If set to No, only the owner of the Report and Users with permission to view others' Reports can see the Report.

Reviewer: ```suggestion
* **Visible for all logged-in Users** - Select **Yes** so that all Users can access the Report. If set to **No**, only the owner of the Report and Users with permission to view others' Reports can see the Report.
```

> * **Owner** - Select the owner of the Report to determine who can see the Report if the 'Visible for all logged-in Users' setting is No.

Reviewer: The 'Visible for all logged-in Users' is how it appears on UI. So, we must stick to the capitalization on UI.

```suggestion
* **Owner** - Select the owner of the Report to determine who can see the Report if the **Visible for all logged-in users** setting is **No**.
```

> * **Visible for all logged-in Users** - Select **Yes** so that all Users can access the Report. If set to **No**, only the owner of the Report and Users with permission to view others' Reports can see the Report.

Reviewer: ```suggestion
* **Visible for all logged-in users** - Select **Yes** so that all Users can access the Report. If set to **No**, only the owner of the Report and Users with permission to view others' Reports can see the Report.
```

* **Published** - Select **Yes** to ensure that the Report is active, ready to send when scheduled in the Report options.
* **Visible for all logged-in Users** - Select **Yes** so that all Users can access the Report. If set to No, only the owner of the Report and Users with permission to view others' Reports can see the Report.
* **Owner** - Select the owner of the Report to determine who can see the Report if the 'Visible for all logged-in Users' setting is No.
* **Dynamic filters settings**

  * Opened by default - Select **Yes** to display dynamic filter options on the Report detail page without requiring Users to open the Filters panel. For example, in an Emails Report, you might want quick access to filters for subject or sent date. The Filters panel only appears when the Report has dynamic filters configured.
  * Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears at the top of the Report table.

> .. vale on

Reviewer: We need to clean up the indentation so nested lists can be clearer about which they belong to.

```suggestion
|

.. vale off

* **Name** - Specify a Report name that makes it easy for you and other Users to identify the purpose of the Report.
* **Data source** - Select the data source appropriate to the Report that you are building. Note that each data source has a different set of available columns, filters, and graphs. The available data sources are:

  #. :doc:`Assets</components/assets>`

     * Assets
     * Asset Downloads

  #. :doc:`Campaigns</campaigns/campaigns_overview>`

     * Campaign Events

  #. :doc:`Channels Messages</queue/message_queue>`

     * Message Queue

  #. :doc:`Emails</channels/emails>`

     * Emails
     * Emails Sent

  #. :doc:`Forms</components/forms>`

     * Forms
     * Form Submissions

  #. :doc:`Contacts</contacts/manage_contacts>`

     * Contacts
     * Multi Touch Attributions
     * First Touch Attributions
     * Last Touch Attributions
     * Contact Point Log
     * Frequency Rules
     * Segment Membership
     * Do Not Contact
     * UTM Codes
     * Group score

  #. :doc:`Companies</companies/companies_overview>`

     * Companies

  #. :doc:`Mobile Notifications</channels/marketing_messages>`

     * Mobile Notifications
     * Mobile Notifications Sent

  #. :doc:`Pages</components/landing_pages>`

     * Landing Pages
     * Page hits

  #. Videos

     * Video hits

  As demonstrated, Mautic displays the data sources in the format: ``<Parent data source> <Child data source>``

  The parent data source provides a high-level summary of the data while the child data source provides a more granular view of Contact behavior that includes your Custom Fields and values.

* **Description - optional** - Add detailed information about the Report, helping you and other Users better understand what the Report is for. You may want to include more information about filters, people, or departments.
* **Published** - Select **Yes** to ensure that the Report is active, ready to send when scheduled in the Report options.
* **Visible for all logged-in Users** - Select **Yes** so that all Users can access the Report. If set to No, only the owner of the Report and Users with permission to view others' Reports can see the Report.
* **Owner** - Select the owner of the Report to determine who can see the Report if the 'Visible for all logged-in Users' setting is No.
* **Dynamic filters settings**

  * Opened by default - Select **Yes** to display dynamic filter options on the Report detail page without requiring Users to open the Filters panel. For example, in an Emails Report, you might want quick access to filters for subject or sent date. The Filters panel only appears when the Report has dynamic filters configured.
  * Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears at the top of the Report table.

.. vale on
```

* **Dynamic filters settings**

>   - Opened by default - Select **Yes** to display dynamic filter options on the Report detail page without requiring Users to open the Filters panel. For example, in an Emails Report, you might want quick access to filters for subject or sent date. The Filters panel only appears when the Report has dynamic filters configured.

Author: Updated Dynamic filters settings documentation based on [PR #16204](https://github.com/mautic/mautic/pull/16204) which moved the date range filter above the Report table and changed the Filters panel to only appear when dynamic filters are configured.

Source: https://github.com/mautic/mautic/pull/16204

>   - Opened by default - Select **Yes** to display dynamic filter options on the Report detail page without requiring Users to open the Filters panel. For example, in an Emails Report, you might want quick access to filters for subject or sent date. The Filters panel only appears when the Report has dynamic filters configured.

Reviewer: There's a vale warning that needs to be addressed: [Mautic.FeatureList] Is this referring to a Mautic feature? If so, use 'Landing Page' instead of 'page'.

>   - Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears above the Report table.

Reviewer: There's a vale warning: [Google.WordList] Use 'preceding' instead of 'above'.
But "above" here isn't the same as "preceding". We can use other word here.

 
```suggestion
  - Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears at the top of the Report table.
```

>   - Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears at the top of the Report table.

Reviewer: ```suggestion
  * Opened by default - Select **Yes** to display dynamic filter options on the Report detail page without requiring Users to open the Filters panel. For example, in an Emails Report, you might want quick access to filters for subject or sent date. The Filters panel only appears when the Report has dynamic filters configured.
  * Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears at the top of the Report table.
```

@promptless-for-oss please address the suggestions.

> 
> * **Dynamic filters settings**
> 
>   * Opened by default - Select **Yes** to display dynamic filter options on the Report detail page without requiring Users to open the Filters panel. For example, in an Emails Report, you might want quick access to filters for subject or sent date. The Filters panel only appears when the Report has dynamic filters configured.
>   * Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears at the top of the Report table.

Reviewer: The word 'page' here is generic. We can wrap the whole list in vale off/on statements.

```suggestion
.. vale off

* **Description - optional** - Add detailed information about the Report, helping you and other Users better understand what the Report is for. You may want to include more information about filters, people, or departments.
* **Published** - Select **Yes** to ensure that the Report is active, ready to send when scheduled in the Report options.
* **Visible for all logged-in Users** - Select **Yes** so that all Users can access the Report. If set to No, only the owner of the Report and Users with permission to view others' Reports can see the Report.
* **Owner** - Select the owner of the Report to determine who can see the Report if the 'Visible for all logged-in Users' setting is No.
* **Dynamic filters settings**

  * Opened by default - Select **Yes** to display dynamic filter options on the Report detail page without requiring Users to open the Filters panel. For example, in an Emails Report, you might want quick access to filters for subject or sent date. The Filters panel only appears when the Report has dynamic filters configured.
  * Hide date range - Select **Yes** to hide the date range filter, preventing Users from changing the dates for the Report data. When visible, the date range filter appears at the top of the Report table.

.. vale on
```

* **Filters** -  Filter the data using conditions and values. This allows the generation of very granular Reports. This option helps you to narrow down the data included in the Report. The data Points used for filters don’t have to be columns that appear in the Report table. A commonly used filter for any Reports that include Contact record data is Email Not Empty, which displays only identified Contacts in the Report. Additional use cases can include Contacts or items that match a certain value, events happening within a certain date range, etc.

.. note::

>    Setting **Dynamic** to **Yes** allows Users to change the data they see in a Report without editing it. They can access the filter from the **Filters** panel at the top of the Report page. The Filters panel only appears when a Report has at least one dynamic filter configured.

Reviewer: There's a vale warning that needs to be addressed: [Mautic.FeatureList] Is this referring to a Mautic feature? If so, use 'Landing Page' instead of 'page'.

>    Setting **Dynamic** to **Yes** allows Users to change the data they see in a Report without editing it. They can access the filter from the **Filters** panel at the top of the Report page. The Filters panel only appears when a Report has at least one dynamic filter configured.

Reviewer: ```suggestion
   .. vale off
   
   Setting **Dynamic** to **Yes** allows Users to change the data they see in a Report without editing it. They can access the filter from the **Filters** panel at the top of the Report page. The Filters panel only appears when a Report has at least one dynamic filter configured.
   
   .. vale on
```
